packages/api/src/router/widgets/calendar.ts

```diff
@@ -0,0 +1,20 @@
+import type { CalendarEvent } from "@homarr/integrations/types";
+
+import { createItemWithIntegrationChannel } from "../../../../redis/src/lib/channel";
+import { createManyIntegrationMiddleware } from "../../middlewares/integration";
+import { createTRPCRouter, publicProcedure } from "../../trpc";
+
+export const calendarRouter = createTRPCRouter({
+  findAllEvents: publicProcedure
+    .unstable_concat(createManyIntegrationMiddleware("sonarr", "radarr", "readarr", "lidarr"))
+    .query(async ({ ctx }) => {
+      return await Promise.all(
+        ctx.integrations.flatMap(async (integration) => {
+          for (const item of integration.items) {
```

Comment:
Shouldn't it just have another parameter for the itemId. Because it makes no sense to return the events for all items of this integration

Reply:
I pushed another commit. Can you look whether this would be the correct approach?
Also, please comment on the TODO